src/DataTable/Pagination.jsx

```diff
@@ -1,27 +1,48 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import Page from './Page.jsx'
+import { ITEMS_PER_PAGE } from '../constants.js'
 
-import Page from './Page'
-
-const Pagination = ({ currentPageNumber, totalNumberOfPages, onChange }) => {
-  const pages =
-    Array
-      .from(Array(totalNumberOfPages).keys())
-      .map(pageNumber => {
-        return <Page
-          key={pageNumber}
-          currentPageNumber={currentPageNumber}
-          pageNumber={pageNumber}
-          onChange={onChange} />
-      })
-
-  if (pages.length <= 1) {
+/**
+ * Calculates how many pages to display in a list
+ * @param {number} rowsFound - The total amount of rows (items)
+ * @param {number} rowsPerPage - How many rows /items are on a single page
+ * @returns {number} - The total amount of pages to display
+ */
+export const calculatePageCount = (rowsFound, rowsPerPage = ITEMS_PER_PAGE) =>
+  rowsPerPage ? Math.ceil(rowsFound / rowsPerPage) : 0
+
+const Pagination = ({ paginationState, totalRowCount }) => {
+  const [currentPageNumber, setCurrentPageNumber] = paginationState
+
+  const pageCount = calculatePageCount(totalRowCount)
+
+  if (pageCount <= 1) {
```

Comment:
Could also have been a ternary.

In general, I prefer avoiding rendering altogether (e.g. `{pageCount && <Pagination />}`, but returning `null` is also fine.